lib/facility.rb

```diff
@@ -1,14 +1,65 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(attributes)
+    @name = attributes[:name]
+    @address = attributes[:address]
+    @phone = attributes[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    vehicle.registration_date = Date.today
+    if vehicle.antique?
+      vehicle.plate_type = :antique
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle? 
+      vehicle.plate_type = :ev
+      @collected_fees += 200
+    else vehicle.plate_type = :regular
+      @collected_fees += 100
+    end
+    @registered_vehicles << vehicle
+  end 
+
+  def eligible_for_written_test?(registrant)
+    registrant.permit? && registrant.age >= 16
+  end
+
+  def administer_written_test(registrant)
+    if eligible_for_written_test?(registrant) == true 
+      registrant.license_data[:written] = true
+    else
+      false
+    end
+  end
+
+  def administer_road_test(registrant)
+    if registrant.license_data[:written]
+      registrant.license_data[:license] = true
+    else
+      false
+    end
+  end
+
+  def renew_license(registrant)
+    if registrant.license_data[:license]
+      registrant.license_data[:renewed] = true
+    else
+      false
```

Comment:
For all these methods, similar to my comment above, you need to check if the service is available at that facility. You could do this with a guard clause at the beginning of the method to avoid a nested if statement:

`return false unless @services.include("License Renewal")`